### ISSUE TITLE
Allow user to add contact by scanning their QR code

### ISSUE THREAD

**Issue by cammellos** [status-im/status-react#14482]
As a user
I would like to add a contact by scanning a QR code

Out of scope:
- Retrieve user info
- User's profile page

### Designs

https://www.figma.com/file/7KIYbhoqNGAIFonE0w9TDz/Messages-for-Mobile?node-id=628%3A174935&t=D6cAP4sUcnKzVSgH-0

Depends on: https://github.com/status-im/status-mobile/issues/14477

**Comment by alwx** [comment]
Should the design for the new QR code reader be done as well? There is a separate issue for it (https://github.com/status-im/status-mobile/issues/13756) but I can take it once this issue is finished/merged and if @siddarthkay doesn't mind + haven't started it yet.

cc @cammellos

**Comment by siddarthkay** [comment]
@alwx : so https://github.com/status-im/status-mobile/issues/13756 is for styling the QR only, i.e having the logo on top of it and I have worked only on that part by implementing that on status-go side here : https://github.com/status-im/status-go/pull/3154.
If you want to work on the QR reader last I remember it was being implemented as part of shell scan screen by @ibrkhalil, so maybe he can provide more input on that one.
